### DISCUSSION TITLE
Setup Requirements Optional/Extras

### DISCUSSION THREAD

**Discussion by cadop** [NVIDIA/warp#4] [Ideas]
Maybe the setup.py could have optional [extras] that automatically put in the usd and urdf, etc. python packages?

**Comment by mmacklin** [comment]
Yes, it could make sense, but at the same time we do not want to impose too many dependencies for users who do not use these features. In the future we may distribute the `warp.sim` package separately to group these together.

**Reply by cadop** [reply]
Sorry if I wasn't clear, I mean to use the special command for extras, so those who just want to use warp can still do `pip install` but others can do `pip install [extras]`

**Reply by cadop** [reply]
Regarding the warp.sim package, I don't have any thoughts on separating it or not, but at least for now it seems like the docs are excluding it?  For example, there are docstrings for `add_body()` but the built doc just says sim is some examples?

**Comment by mmacklin** [comment]
I see, seems like a good idea, I will look into it. 

Thanks, 
Miles
